sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -204,6 +204,29 @@ case class Elt(children: Seq[Expression])
       }
     }
   }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val index = children.head.map(_.genCode(ctx))(0)
+    val strings = children.tail.map(_.genCode(ctx))
+    val stringValues = strings.map { eval =>
+      s"${eval.isNull} ? null : ${eval.value}"
+    }.mkString(", ")
+    val indexVal = ctx.freshName("index")
+    val stringArray = ctx.freshName("strings");
+
+    ev.copy(index.code + "\n" + strings.map(_.code).mkString("\n") + s"""
+      int $indexVal = ${index.value} - 1;
+      UTF8String[] $stringArray = {$stringValues};
```

Comment:
Because the index is only evaluated runtime, I think we can' just evaluate the string we need? Or we can wrap these string expressions in an if/else if or switch block, then we can only evaluate the string expression specified by the index. What do you think?